numba/pythonapi.py

```diff
@@ -1581,3 +1582,65 @@ def string_from_constant_string(self, string):
         cstr = self.context.insert_const_string(self.module, string)
         sz = self.context.get_constant(types.intp, len(string))
         return self.string_from_string_and_size(cstr, sz)
+
+    def call_jit_code(self, func, sig, args):
+        """Calls into Numba jitted code and propagate error using the Python
+        calling convention.
+
+        Parameters
+        ----------
+        func : function
+            The Python function to be compiled. This function is compiled
+            in nopython-mode.
+        sig : numba.typing.Signature
+            The function signature for *func*.
+        args : Sequence[llvmlite.binding.Value]
+            LLVM values to use as arguments.
+
+        Returns
+        -------
+        (is_error, res) :  2-tuple of llvmlite.binding.Value.
+            is_error : true iff *func* raised an exception.
+            res : Returned value from *func* iff *is_error* is false.
+
+        If *is_error* is true, this method will adapt the nopython exception
+        into a Python exception. Caller should return NULL to Python to
+        indicate an error.
```

Comment:
An example would be quite big because it requires so much context. In that case, I think it's better to just find uses of this method in the source tree.

Reply:
o.k.